### PR TITLE
simplify file layout for tccp resource

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/controller/v25/resource/tccp/main_stack.go [hunk 1]
```diff
@@ -1,71 +0,0 @@
...
-	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
-	"github.com/giantswarm/microerror"
-
-	"github.com/giantswarm/aws-operator/service/controller/v25/adapter"
-	"github.com/giantswarm/aws-operator/service/controller/v25/controllercontext"
-	"github.com/giantswarm/aws-operator/service/controller/v25/key"
-	"github.com/giantswarm/aws-operator/service/controller/v25/templates"
-)
-
-func (r *Resource) getMainGuestTemplateBody(ctx context.Context, customObject v1alpha1.AWSConfig, stackState StackState) (string, error) {
```
Comment (xh3b4sd) [inline service/controller/v25/resource/tccp/main_stack.go]: There is nothing like a main template anymore. There is only the stack template now.

File: service/controller/v25/resource/tccp/main_stack_test.go [hunk 2]
```diff
@@ -1,619 +0,0 @@
...
-	cc := controllercontext.Context{}
-	ctx = controllercontext.NewContext(ctx, cc)
-
-	_, err = newResource.getMainGuestTemplateBody(ctx, customObject, StackState{})
-	if err == nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-}
-
-func TestMainGuestTemplateExistingFields(t *testing.T) {
```
Comment (xh3b4sd) [inline service/controller/v25/resource/tccp/main_stack_test.go]: Dropping the tests as they turned out to be not too useful. I am planning to move the adapters left over to the tccp templates and have rendering tests there.

File: service/controller/v25/resource/tccp/create_test.go [hunk 3]
```diff
@@ -126,3 +126,25 @@ func Test_Resource_Cloudformation_newCreate(t *testing.T) {
 		})
 	}
 }
+
+func statusWithAllocatedSubnet(cidr string, azs []string) v1alpha1.AWSConfigStatus {
```
Comment (xh3b4sd) [inline service/controller/v25/resource/tccp/create_test.go]: That is apparently still used for other tests so just keeping this for now.

File: service/controller/v25/resource/tccp/resource.go [hunk 4]
```diff
@@ -102,6 +106,64 @@ func (r *Resource) getCloudFormationTags(customObject v1alpha1.AWSConfig) []*aws
 	return awstags.NewCloudFormation(tags)
 }
 
+func (r *Resource) newTemplateBody(ctx context.Context, customObject v1alpha1.AWSConfig, stackState StackState) (string, error) {
```
Comment (xh3b4sd) [inline service/controller/v25/resource/tccp/resource.go]: Just moving it to here for now.

File: service/controller/v25/resource/tccp/main_stack_test.go (hunk 2 above)
Comment (MarcelMue) [reply service/controller/v25/resource/tccp/main_stack_test.go]: Why are they not useful? I don't quite get this.

Comment (xh3b4sd) [reply service/controller/v25/resource/tccp/main_stack_test.go]: Their design is wrong. They cause a lot of maintenance. They do not really assure service quality. The rendering will be rewritten anyway so all tests will just become outdated eventually. I want to cleanup all that old crap that stands in our way.

**Comment by xh3b4sd** [comment]
I have this covered.

**Pull request merged**